### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by fivegoblins**

### CODE REVIEW

File: basic-javascript/project-1.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
 /* eslint-disable arrow-parens */
 
 const multiplyByTen = num => {
+  if (!num) {
+    return 0;
+  }
+  if (typeof num === 'string') {
```
Comment (gooseandmegander) [inline basic-javascript/project-1.js:7]: What if a boolean is supplied to multiplyByTen?

File: basic-javascript/__tests__/project-1.spec.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        it('should subtract five from the given number', () => {
+            const actual = helpers.subtractFive(75);
+            expect(actual).toEqual(70);
+        });
+
+        it('should return NaN when called with a string', () => {
+            const actual = helpers.subtractFive();
+            expect(actual).toBe(NaN);
+        });
+    });
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:37]: In a hypothetical context, could there be an edge case for negative numbers?

File: basic-javascript/__tests__/project-1.spec.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        it('should return true if strings are same length', ()=> {
+            const actual = helpers.areSameLength('hello', 'howdy');
+            expect(actual).toBeTruthy();
+        });
+
+        it('should return false if strings are not same length', ()=> {
+            const actual = helpers.areSameLength('good morning', 'sup');
+            expect(actual).toBeFalsy();
+        });
+    });
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:49]: What if `true` or `false` are supplied?

File: basic-javascript/__tests__/project-1.spec.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        it('should return false if number is greater than 90', ()=> {
+            const actual = helpers.lessThanNinety(91);
+            expect(actual).toBeFalsy();
+        });
+
+        it('should return false if number is equal to 90', ()=> {
+            const actual = helpers.lessThanNinety(90);
+            expect(actual).toBeFalsy();
+        });
+    });
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:78]: What if a string is supplied?

File: basic-javascript/__tests__/project-1.spec.js [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+    describe('add()', ()=> {
+        it('should return the sum of given numbers', ()=> {
+            const actual = helpers.add(4, -12);
+            expect(actual).toEqual(-8);
+        });
+    });
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:102]: What if a string is one of the arguments?

File: basic-javascript/__tests__/project-1.spec.js [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+            expect(actual).toEqual(48);
+        });
+    });
+
+    describe('divide()', ()=> {
+        it('should return the quotient of given numbers', ()=> {
+            const actual = helpers.divide(12, 4);
+            expect(actual).toEqual(3);
+        });
+    });
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:123]: What should return if the denominator is 0?

File: basic-javascript/__tests__/project-1.spec.js [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+
+        it('should round up to the nearest whole integer', ()=> {
+            const actual = helpers.roundNumber(3.5);
+            expect(actual).toBe(4);
+        });
+    });
+
+    describe('addExclamationPoint()', ()=> {
+        it('should add an exclamation point to the end of the string', ()=> {
+            const actual = helpers.addExclamationPoint('hello');
```
Comment (gooseandmegander) [inline basic-javascript/__tests__/project-1.spec.js:213]: Numbers are also valid arguments the way the test is set up now.

File: basic-javascript/project-2.js [hunk 8]
```diff
@@ -105,6 +119,9 @@ const wordsToSentence = words => {
       newSentence += ` ${words[i]}`;
     }
   }
+  if (typeof words === 'number') {
+    return null;
+  }
```
Comment (gooseandmegander) [inline basic-javascript/project-2.js:124]: Consider moving your if condition to execute before the word appending functionality happens. Something like this would work:
```
for (let i = 0; i < words.length; i++) {
    if (typeof words[i] === 'number') return null;
    if (i === 0) {
      newSentence += `${words[i]}`;
    } else {
      newSentence += ` ${words[i]}`;
    }
  }
```

**Review by gooseandmegander — COMMENTED** [review]
# Objectives
* Familiarity with writing unit tests for JavaScript code.

# MVP
- [x] project-1.spec.js tests passing
- [x] project-2.spec.js tests passing

# Stretch
- [ ] arrays.spec.js tests passing
- [ ] objects.spec.js tests passing

# Great
All the tests pass except for one. See my comment below for that. There are both "red" and "green" tests.

# Requests for Improvement
* Check out other types of testing.
* Use something new from the docs you haven't used yet to generate tests.
* Think about edge cases.

# Rating: 2
